### PR TITLE
Optimize behavior for rare use cases (`serve -e prod` & `generate -e dev`)

### PR DESCRIPTION
**Pull request by nighca**
Optimize behavior for rare use cases including:

1. `fec-builder serve -e production`
2. `fec-builder generate -e development`

### `serve -e production`

This is meaningful for situations like:

* Check production-build result locally

Here is an example:

wecom-temp-2ab9c757c131a9bc1c9b0a16c2bca5f7

### `generate -e development`

This is meaningful for situations like:

* Debug application in staging / preview environment
* Debug builder features like tree shaking

Here is a sample for generate result:

```js

/***/ "./src/App.tsx":
/*!*********************!*\
  !*** ./src/App.tsx ***!
  \*********************/
/***/ ((__unused_webpack_module, __webpack_exports__, __webpack_require__) => {

__webpack_require__.r(__webpack_exports__);
/* harmony export */ __webpack_require__.d(__webpack_exports__, {
/* harmony export */   "default": () => (/* binding */ App)
/* harmony export */ });
/* harmony import */ var react__WEBPACK_IMPORTED_MODULE_0__ = __webpack_require__(/*! react */ "./node_modules/react/index.js");
/* harmony import */ var _logo_svg__WEBPACK_IMPORTED_MODULE_1__ = __webpack_require__(/*! ./logo.svg */ "./src/logo.svg");
/* harmony import */ var _Name__WEBPACK_IMPORTED_MODULE_2__ = __webpack_require__(/*! ./Name */ "./src/Name/index.tsx");
var _s = $RefreshSig$();




function App() {
  _s();

  const [name, setName] = (0,react__WEBPACK_IMPORTED_MODULE_0__.useState)('');
  (0,react__WEBPACK_IMPORTED_MODULE_0__.useEffect)(() => {
    __webpack_require__.e(/*! import() */ "src_extra_ts").then(__webpack_require__.bind(__webpack_require__, /*! ./extra */ "./src/extra.ts")).then(extra => {
      setName(extra.name);
    });
  }, []);
  return /*#__PURE__*/react__WEBPACK_IMPORTED_MODULE_0__.createElement("div", {
    className: "App"
  }, /*#__PURE__*/react__WEBPACK_IMPORTED_MODULE_0__.createElement(_logo_svg__WEBPACK_IMPORTED_MODULE_1__["default"], {
    style: {
      width: '100px'
    }
  }), "Hello", /*#__PURE__*/react__WEBPACK_IMPORTED_MODULE_0__.createElement(_Name__WEBPACK_IMPORTED_MODULE_2__["default"], {
    name: name
  }), "!");
}

_s(App, "nVbhOIYDoHElhkRqYpJCqH0DXNE=");

_c = App;

var _c;

$RefreshReg$(_c, "App");

/***/ }),

/***/ "./src/Name/index.tsx":
/*!****************************!*\
  !*** ./src/Name/index.tsx ***!
  \****************************/
/***/ ((__unused_webpack_module, __webpack_exports__, __webpack_require__) => {

__webpack_require__.r(__webpack_exports__);
/* harmony export */ __webpack_require__.d(__webpack_exports__, {
/* harmony export */   "default": () => (/* binding */ Name)
/* harmony export */ });
/* harmony import */ var react__WEBPACK_IMPORTED_MODULE_0__ = __webpack_require__(/*! react */ "./node_modules/react/index.js");
/* harmony import */ var _style_m_less__WEBPACK_IMPORTED_MODULE_1__ = __webpack_require__(/*! ./style.m.less */ "./src/Name/style.m.less");


function Name({
  name
}) {
  return /*#__PURE__*/react__WEBPACK_IMPORTED_MODULE_0__.createElement("span", {
    className: _style_m_less__WEBPACK_IMPORTED_MODULE_1__["default"].name
  }, name);
}
_c = Name;

var _c;

$RefreshReg$(_c, "Name");
```

### CODE REVIEW

File: src/serve.ts [hunk 1]
```diff
@@ -10,17 +10,22 @@ import WebpackDevServer from 'webpack-dev-server'
...
 
 // 业务项目的配置文件，变更时需要重启 server
 const projectConfigFiles = [
   'tsconfig.json'
 ]
 
 async function serve(port: number) {
+  // 对于 dev server，非 Dev 的 env 取值是没有意义的，
+  // 因此这里强制指定为 env: Dev，避免出现令人困惑的行为
+  setEnv(Env.Dev)
```
Comment (liaoyu) [inline src/serve.ts:27]: nip: 我之前倒是用过起 `dev server` 的情况下开启 `production` 模式，用于本地验证 `portal` 站点的打包结果，可以考虑保留~

Comment (nighca) [reply src/serve.ts]: done https://github.com/Front-End-Engineering-Cloud/builder/pull/163/commits/83585b5ac6bcce2c5eabc882da889c808fc34b33 调整了命令 `serve` 对应的行为

**Review by liaoyu — APPROVED** [review]
🐮

**Review by liaoyu — APPROVED** [review]
lgtm

**Pull request merged**